### PR TITLE
Header component for transaction confirmation screens

### LINKED ISSUE #15222: Create a new network + account + balance header component for transaction confirmation screens
## Resources
[Figma Design file](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=2007%3A19831)

## Acceptance Criteria

1. This component displays the relevant network, account and balance for the transaction as per design
Screen Shot 2022-07-13 at 14 47 05

2. For token allowance screens the balance displayed should always be the user's balance for the specific token the user is being requested allowance for - in the design example it's `DAI`. On other confirmation screen, this might vary.
The token allowance screens are shown on the Figma screenshot below:
image

### PR DESCRIPTION
**Pull request by filipsekulic**
## Explanation

Created a header component for transaction confirmation screens (network + account + balance). The component will be implemented and used in the future. However, it can be seen now within the storybook. This component has following properties:

- networkName (type: string)
- accountName (type: string)
- accountBalance (type: number)
- tokenName (type: string)
- accountAddress (type: string)

## More Information
* Fixes #15222 

## Screenshots/Screencaps
Screenshot 2022-08-17 at 17 35 31
Screenshot 2022-08-17 at 17 35 55

## Manual Testing Steps
The component can be seen and tested from the storybook (`yarn storybook`). Then go to the `COMPONENTS` tab and open it from the `APP` tab - `NetworkAccountBalanceHeader`.

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

File: ui/components/app/network-account-balance-header/network-account-balance-header.stories.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import NetworkAccountBalanceHeader from './network-account-balance-header';
+
+export default {
+  title: 'Components/APP/NetworkAccountBalanceHeader',
```
Comment (NidhiKJha) [inline ui/components/app/network-account-balance-header/network-account-balance-header.stories.js:5]:
```suggestion
  title: 'Components/App/NetworkAccountBalanceHeader',
```

**Review by NidhiKJha — DISMISSED** [review]
LGTM! Just a minor change suggestion

File: ui/components/app/network-account-balance-header/network-account-balance-header.stories.js (hunk 1 above)
Comment (filipsekulic) [reply ui/components/app/network-account-balance-header/network-account-balance-header.stories.js]: Fixed [here](0ee04f39cd190e81aa1961642fe7a33124071b2a)

**Review by brad-decker — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Comment by SaraCheikh** [comment]
Hello, some details to adapt this to our existing components and tokens:

- The badge seems to have a grey border that the component `Avatar with Badge` in the[ Figma file](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=1942%3A13595) doesn't have.
- The font of the title (network and balance) seems to be 12px in medium which doesn't exist in our text tokens. for 12px we have the token `S-Body-SM` in regular.  For this type of title I would advice to use at least the baseline token (14px) since 12px is quite small. 

Screenshot 2022-09-09 at 14 15 07

**Comment by filipsekulic** [comment]
> Hello, some details to adapt this to our existing components and tokens:
> 
> * The badge seems to have a grey border that the component `Avatar with Badge` in the[ Figma file](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=1942%3A13595) doesn't have.
> * The font of the title (network and balance) seems to be 12px in medium which doesn't exist in our text tokens. for 12px we have the token `S-Body-SM` in regular.  For this type of title I would advice to use at least the baseline token (14px) since 12px is quite small.
> 
> Screenshot 2022-09-09 at 14 15 07

Hey @SaraCheikh!
Good catch! Thank you for pointing this out.
I made these changes in this PR: https://github.com/MetaMask/metamask-extension/pull/15727

**Pull request merged**